nav2_mppi_controller/src/controller.cpp

```diff
@@ -122,6 +130,36 @@ void MPPIController::setSpeedLimit(const double & speed_limit, const bool & perc
   optimizer_.setSpeedLimit(speed_limit, percentage);
 }
 
+void MPPIController::publish_optimal_path(const xt::xtensor<float, 2> & optimal_traj)
+{
+  auto & size = optimal_traj.shape()[0];
+  if (!size) {
+    return;
+  }
+
+  nav_msgs::msg::Path optimal_path;
+  optimal_path.header.stamp = clock_->now();
+  optimal_path.header.frame_id = (costmap_ros_->getGlobalFrameID()).c_str();
+
+  for (size_t i = 0; i < size; i++) {
+    // create new pose for the path
+    geometry_msgs::msg::PoseStamped pose_stamped;
+    pose_stamped.header.frame_id = (costmap_ros_->getGlobalFrameID()).c_str();
+
+    // position & orientation
+    pose_stamped.pose = utils::createPose(optimal_traj(i, 0), optimal_traj(i, 1), 0.);
+
+    tf2::Quaternion quaternion_tf2;
+    quaternion_tf2.setRPY(0., 0., optimal_traj(i, 2));
+    auto quaternion = tf2::toMsg(quaternion_tf2);
+    pose_stamped.pose.orientation = quaternion;
```

Comment:
Can simply be `pose_stamped.pose.orientation = tf2::toMsg(quaternion_tf2);`, don't need intermediate variable